src/test/java/ValidUrlsTest.java

```diff
@@ -0,0 +1,90 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+import io.jenkins.plugins.designlibrary.UISample;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import org.htmlunit.html.DomNode;
+import org.htmlunit.html.HtmlAnchor;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.junit.jupiter.WithJenkins;
+
+@WithJenkins
+@TestInstance(PER_CLASS)
+class ValidUrlsTest {
+
+    private JenkinsRule jenkins;
+
+    private List<UISample> samples;
+
+    @BeforeAll
+    void beforeAll(JenkinsRule jenkins) {
+        this.jenkins = jenkins;
+        this.samples = jenkins.getInstance().getExtensionList(UISample.class);
+    }
+
+    /**
+     * Validate that relative URLs on a page actually link to a {@link UISample}
+     */
+    @ParameterizedTest
+    @MethodSource("getPages")
+    void validUrls(String url) throws Exception {
+        try (var webClient = jenkins.createWebClient().withJavaScriptEnabled(false)) {
+            // We get a bunch of spam in our logs about missing CSS, let's ignore that
+            webClient.getOptions().setPrintContentOnFailingStatusCode(false);
+            webClient.getOptions().setThrowExceptionOnFailingStatusCode(false);
+
+            List<String> validUrls =
+                    new ArrayList<>(samples.stream().map(UISample::getUrlName).toList());
+
+            var page = webClient.goTo(url);
+            var links = page.querySelectorAll(".jdl-section a");
+
+            if (links.isEmpty()) {
+                System.out.println("🤷 No URLs on this page");
```

Comment:
can you include what page it is

Reply:
Was hoping the test name would be outputted in a nice manner in the output, but nope :/ Done.